Travis adding py3.5

bsipocz opened the pull request:
This PR updates the matrix to test against py 3.5 and numpy 1.10, too.

bsipocz commented:
I expect some fails as there isn't numpy 1.10 on conda for python 3.3 and 2.6. I'll add a temporary commit to have these tested against 1.9 instead, and we'll update again once there will be builds available in ``astropy-ci-extras`` channel.

astrofrog commented:
To be honest, I think we can probably drop support for Python 2.6 - I know that in the core package, people want to do this pretty slowly (i.e. announce it one release ahead, etc.) but in affiliated packages like this, I think we can just support when we decide.

larrybradley commented:
+1 for dropping python 2.6.

astrofrog commented on docs/photutils/getting_started.rst:
```diff
@@ -94,7 +94,7 @@ apertures:
     hdu = datasets.load_star_image()
     image = hdu.data[500:700, 500:700]
-    image -= np.median(image)
+    image = image.astype(float) - np.median(image)
```
I wonder whether it would clearer to leave this line as before and instead do ``image = hdu.data[...].astype(float)``

bsipocz replied:
Done

bsipocz commented:
OK, so I've removed 2.6 support and updated CHANGES, too.

However updating the appveyor was a bit over ambitious, and now it's failing with python 3.5&np 1.10. 
Does any of you have ideas how to fix it?
